presto-docs/src/main/sphinx/release/release-0.289.rst

```diff
@@ -0,0 +1,72 @@
+* ... Add :doc:`/clients/superset` documentation :pr:`23194`.
+* Make map_top_n function more deterministic by using keys to break ties.
+* Pass full session to avoid Unknown connector errors :pr:`22936`.
+* Switch default Parquet writer version to PARQUET_1_0. Please set the hive config `hive.parquet.writer.version = PARQUET_2_0` for old behavior. :pr:`23369`.
+* The Iceberg connector when configured with Hive, can now configure table metadata refresh timeouts. See documentation for details.
+* Update Joda-Time to 2.12.7 to use 2024a tzdata. Note: a corresponding update to the Java runtime should also be made to ensure consistent timezone data. For example, Oracle JDK 8u381, tzdata2024a rpm for OpenJDK, or use Timezone Updater Tool to apply 2024a tzdata to existing JVM. :pr:`23027`.
+* Upgrade Airlift to 0.215 :pr:`23356`.
+* Upgrade avro to 1.11.3 due CVE-2023-39410 :pr:`23142`.
+* Upgrade guava to 32.1.0-jre due CVE-2023-2976 :pr:`23127`.
+* Upgrade json-path to 2.9.0 due CVE-2023-1370 :pr:`23104`.
+
+Hive Connector Changes
+______________________
+* Fix filtering by info columns $file_size and $file_modified_time, which were ignored before. :pr:`23411`.
+* Fix hash calculation for Timestamp column to be hive compatible when writing to a table bucketed by Timestamp.  :pr:`22890`.
+* Add config `hive.legacy-timestamp-bucketing` and session property ``hive.legacy_timestamp_bucketing`` to use the original hash function for Timestamp column, which is not hive compatible. :pr:`22890`.
+* Add support for setting the max size in bytes for directory listing cache. This can be set via the new `hive.file-status-cache.max-retained-size` configuration property. `hive.file-status-cache-size` is now deprecated :pr:`23176`.
+* Add support to skip empty files using configuration property ``hive.skip_empty_files`` :pr:`22727`.
+
+Iceberg Connector Changes
+_________________________
+* Add procedure `remove_orphan_files` to remove orphan files that are not referenced in any metadata files for Iceberg. :pr:`23267`.
+* Add table properties `metadata_previous_versions_max` and `metadata_delete_after_commit` to maintain the previous metadata files :pr:`23260`.
+* Enable Iceberg with hive catalog to support deleting old metadata files after commit based on the table properties :pr:`23260`.
+
+Verifier Changes
+________________
+* Add `control.reuse-table` and `test.reuse-table` configuration properties for the Presto Verifier to reuse the output tables of the source query for control and test.
+* Add verifier config --validate-string-as-double to control applying floating point validation to the column composed of varchar, if it is detected that the varchar column is derived from casting floating points. :pr:`23312`.
+
+SPI Changes
+___________
+* Add ``equalValuesAreIdentical`` to ``Type``.  Override this method to return ``false`` when the values of the type may have more than one canonical representation.
```

Comment:
This reads a little wonky. Canonical representations should be unique and if they're not they're not a canonical representation.